examples/experimental/dagster-airlift/dagster_airlift/in_airflow/proxying_fn.py

```diff
@@ -38,36 +39,52 @@ def proxying_to_dagster(
     if not logger:
         logger = logging.getLogger("dagster_airlift")
     logger.debug(f"Searching for dags proxied to dagster{suffix}...")
-    proxying_dags: List[DAG] = []
+    task_proxying_dags: List[DAG] = []
+    override_dags: List[DAG] = []
     all_dag_ids: Set[str] = set()
     # Do a pass to collect dags and ensure that proxied information is set correctly.
     for obj in global_vars.values():
         if not isinstance(obj, DAG):
             continue
         dag: DAG = obj
         all_dag_ids.add(dag.dag_id)
-        if not proxied_state.dag_has_proxied_state(dag.dag_id):
+        if not proxied_state.dag_proxies_dag(dag.dag_id) and not proxied_state.dag_proxies_tasks(
+            dag.dag_id
+        ):
             logger.debug(f"Dag with id `{dag.dag_id}` has no proxied state. Skipping...")
             continue
         logger.debug(f"Dag with id `{dag.dag_id}` has proxied state.")
         proxied_state_for_dag = proxied_state.dags[dag.dag_id]
-        for task_id in proxied_state_for_dag.tasks.keys():
-            if task_id not in dag.task_dict:
-                raise Exception(
-                    f"Task with id `{task_id}` not found in dag `{dag.dag_id}`. Found tasks: {list(dag.task_dict.keys())}"
-                )
-            if not isinstance(dag.task_dict[task_id], BaseOperator):
-                raise Exception(
-                    f"Task with id `{task_id}` in dag `{dag.dag_id}` is not an instance of BaseOperator. This likely means a MappedOperator was attempted, which is not yet supported by airlift."
-                )
-        proxying_dags.append(dag)
+        if proxied_state_for_dag.proxied:
+            logger.debug("Full dag-level override detected.")
+            override_dags.append(dag)
+        else:
+            logger.debug("Task-level overrides detected.")
+            for task_id in proxied_state_for_dag.tasks.keys():
+                if task_id not in dag.task_dict:
+                    raise Exception(
+                        f"Task with id `{task_id}` not found in dag `{dag.dag_id}`. Found tasks: {list(dag.task_dict.keys())}"
+                    )
+                if not isinstance(dag.task_dict[task_id], BaseOperator):
+                    raise Exception(
+                        f"Task with id `{task_id}` in dag `{dag.dag_id}` is not an instance of BaseOperator. This likely means a MappedOperator was attempted, which is not yet supported by airlift."
+                    )
+        task_proxying_dags.append(dag)
 
     if len(all_dag_ids) == 0:
         raise Exception(
             "No dags found in globals dictionary. Ensure that your dags are available from global context, and that the call to `proxying_to_dagster` is the last line in your dag file."
         )
 
-    for dag in proxying_dags:
+    for dag in override_dags:
```

Comment:
impl is pretty straightforward. Went with a single task overriding the whole task structure.

Reply:
wdym by smashing?